app/javascript/react/screens/App/Plan/Plan.js

```diff
@@ -65,11 +65,11 @@ class Plan extends React.Component {
         const mostRecentRequest = getMostRecentRequest(miq_requests);
         const planRequestId = mostRecentRequest.id;
         fetchPlanRequestAction(fetchPlanRequestUrl, planRequestId);
-        if (mostRecentRequest.status === 'active') {
-          this.startPolling(planRequestId);
-        } else if (
-          mostRecentRequest.request_state === 'finished' ||
-          mostRecentRequest.status === 'Error'
+        this.startPolling(planRequestId);
```

Comment:
@mturley Calling `this.startPolling(planRequestId)` was a good call, because it does prove that if polling is called regardless of the value in `mostRecentRequest.status`, then the page does refresh -- I confirmed this on an appliance.

I think this whole if/else block can be replaced and simplified by this --
```js
if (mostRecentRequest.fulfilled_on === null) {
  this.startPolling(planRequestId);
} else {
  this.setState(() => ({
    planFinished: true
  }));
}

```

`fulfilled_on` being `null` is a reliable indicator of the request being in it's in-progress state.

Reply:
Will test this some more tomorrow. Thanks!

Reply:
Scratch that, if we have `mostRecentRequest.id` we will have `mostRecentRequest.fulfilled_on`. I'll go with your version @AparnaKarve